subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/ResourceManagementFilter.java

```diff
@@ -97,9 +102,43 @@ public void doFilter( final ServletRequest request, final ServletResponse respon
             ThreadContext.clearContext();
 
             logger.info( "END request: {} (from: {})", tn, clientAddr );
+
+            MDC.clear();
         }
     }
 
+    public static final String HTTP_REQUEST_EXTERNAL_ID = "http-request-external-id";
+    public static final String HTTP_REQUEST_INTERNAL_ID = "http-request-internal-id";
+    public static final String HTTP_REQUEST_PREFERRED_ID = "http-request-preferred-id";
+
+    /**
+     * Put to MDC / threadContext request IDs.
+    */
+    private void putRequestIDs( HttpServletRequest hsr, ThreadContext threadContext )
+    {
+        /* We would always generate internalID and provide that in the MDC.
+         * If the calling service supplies an externalID, we'd map that under its own key.
+         * PreferredID should try to use externalID if it's available, and default over to using internalID if it's not.
+         * What this gives us is a single key we can use to reference an ID for the request,
+         * and whenever possible it'll reflect the externally supplied ID.
+         */
+        String internalID = UUID.randomUUID().toString();
+        String externalID = hsr.getHeader( HTTP_REQUEST_EXTERNAL_ID );
```

Comment:
This is nullable, so it'll need a null-check before setting in the MDC or the ThreadContext. That's probably why the build is failing.